### PR TITLE
Add IP address to RPC errors

### PR DESCRIPTION
**Pull request by acodrington**
# Background

We would like to have more information in our logs to help us diagnose potential future issues. This PR adds the relevant IP address to a number of RPC error logs.

# Results
Fixes [sc-60356]

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [x] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/master/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [x] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [x] I have considered appropriate testing for my change.

### CODE REVIEW

File: source/Halibut/Transport/SecureListener.cs [hunk 1]
```diff
@@ -164,7 +165,7 @@ void WaitForPendingConnectionOrCancellation()
                     catch (Exception ex)
                     {
                         numberOfFailedAttemptsInRow++;
-                        log.WriteException(EventType.ErrorInInitialisation, "Error accepting TCP client", ex);
+                        log.WriteException(EventType.ErrorInInitialisation, "Error accepting TCP client: {0}", ex, client?.Client.RemoteEndPoint);
```
Comment (nathanwoctopusdeploy) [inline source/Halibut/Transport/SecureListener.cs:168]: Client.RemoteEndPoint can throw exceptions so do you think it would be safer to call an extension method or similar that wraps the call in a try catch so we guarantee we don't error trying to log an error 
image

Comment (nathanwoctopusdeploy) [reply source/Halibut/Transport/SecureListener.cs:168]: I don't know if it would ever throw in the situation we are using it, but just incase

Comment (nathanwoctopusdeploy) [reply source/Halibut/Transport/SecureListener.cs:168]: Kind of a on th lines of Halibut.Transport.Proxy.Utils.GetHost and GetPort

File: source/Halibut/Transport/TcpClientExtensions.cs [hunk 2]
```diff
@@ -93,5 +94,17 @@ public static void CloseImmediately(this TcpClient client, Action<Exception> onE
...
+        public static string GetRemoteEndpointString(this TcpClient client)
+        {
+            try
+            {
+                return client?.Client.RemoteEndPoint?.ToString() ?? "<none>";
+            }
+            catch
+            {
+                return "<error>";
+            }
```
Comment (acodrington) [inline source/Halibut/Transport/TcpClientExtensions.cs:107]: RemoteEndPoint can throw if the socket has already been disposed/closed, but that's a bit wordy for this usage i.e. log message.

I figured `<error>` vs `<none`> would help us differentiate via the logs what happened without being too wordy. These will be easy to find via a quick CTRL + F.

File: source/Halibut/Transport/SecureListener.cs [hunk 3]
```diff
@@ -202,29 +203,29 @@ async Task HandleClient(TcpClient client)
...
+                log.WriteException(EventType.ErrorInInitialisation, "Error initializing TCP client: {0}", ex, client.GetRemoteEndpointString());
             }
         }
 
         async Task ExecuteRequest(TcpClient client)
         {
             var connectionAuthorizedAndObserved = false;
 
-            var clientName = client.Client.RemoteEndPoint;
+            string clientName = client.GetRemoteEndpointString();
```
Comment (nathanwoctopusdeploy) [inline source/Halibut/Transport/SecureListener.cs:222]: var -> string?

Comment (acodrington) [reply source/Halibut/Transport/SecureListener.cs:222]: Unintentional, will revert

File: source/Halibut/Transport/SecureWebSocketListener.cs [hunk 4]
```diff
@@ -114,7 +115,7 @@ async Task Accept()
                     {
                         if (!cts.IsCancellationRequested)
                         {
-                            log.WriteException(EventType.ErrorInInitialisation, "Error accepting Web Socket client", ex);
+                            log.WriteException(EventType.ErrorInInitialisation, "Error accepting Web Socket client: {0}", ex, context?.Request.RemoteEndPoint);
```
Comment (nathanwoctopusdeploy) [inline source/Halibut/Transport/SecureWebSocketListener.cs:118]: Should these use the extension method or is it a different APi that doesn't throw

**Review by nathanwoctopusdeploy — APPROVED** [review]

File: source/Halibut/Transport/SecureWebSocketListener.cs (hunk 4 above)
Comment (acodrington) [reply source/Halibut/Transport/SecureWebSocketListener.cs:118]: The latter - it doesn't throw but I just checked and it _can_ be null. I'm not sure what the log method will do with a null object, so I'll check that behaviour and update the code if necessary.

**Comment by acodrington** [comment]
[sc-60356]
